src/main/scala/com/typesafe/sbt/web/SbtWeb.scala

```diff
@@ -318,22 +304,50 @@ object SbtWeb extends AutoPlugin {
     nodeModules := nodeModuleGenerators(_.join).map(_.flatten).value
   )
 
+  def webJarsPathPrefix: Def.Initialize[Task[String]] = Def.task {
+    path(s"$WEBJARS_PATH_PREFIX/${moduleName.value}/${version.value}/")
+  }
+
+  def syncExportedAssets(track: TrackLevel): Def.Initialize[Task[File]] = Def.task {
+    val syncTargetDir = classDirectory.value
+    val syncRequired = TrackLevel.intersection(track, exportToInternal.value) match {
+      case TrackLevel.TrackAlways =>
+        true
+      case TrackLevel.TrackIfMissing | TrackLevel.NoTracking =>
+        !(syncTargetDir / webJarsPathPrefix.value).exists()
+    }
+    if (syncRequired) {
+      state.value.log.info(s"Exporting ${configuration.value}:${moduleName.value}")
+      syncMappings(streams.value.cacheDirectory, exportedMappings.value, syncTargetDir)
+    } else
+      syncTargetDir
+  }
```

Comment:
Hmm. Ok. I'll test this out further with a transpiler in there - probably coffee script because it is known to me.

@pvlugter: I'll need to pick your brains re the test products...

> On 17 Apr 2016, at 09:14, eugene yokota <notifications@github.com> wrote:
> 
> In src/main/scala/com/typesafe/sbt/web/SbtWeb.scala:
> 
> > +  }
> > +
> > +  def syncExportedAssets(track: TrackLevel): Def.Initialize[Task[File]] = Def.task {
> > +    val syncTargetDir = classDirectory.value
> > +    val syncRequired = TrackLevel.intersection(track, exportToInternal.value) match {
> > +      case TrackLevel.TrackAlways =>
> > +        true
> > +      case TrackLevel.TrackIfMissing | TrackLevel.NoTracking =>
> > +        !(syncTargetDir / webJarsPathPrefix.value).exists()
> > +    }
> > +    if (syncRequired) {
> > +      state.value.log.info(s"Exporting ${configuration.value}:${moduleName.value}")
> > +      syncMappings(streams.value.cacheDirectory, exportedMappings.value, syncTargetDir)
> > +    } else
> > +      syncTargetDir
> > +  }
> Doesn't sbt-web invoke the source generator on this line:
> 
> managedSources := sourceGenerators(_.join).map(_.flatten).value
> and doesn't mapping in sources here cause dependency to that?:
> 
> sources := managedSources.value ++ unmanagedSources.value,
> mappings in sources <<= relativeMappings(sources, sourceDirectories),
> Alternative Javascript plugins are hooking into sourceGenerators, aren't they not?
> 
> —
> You are receiving this because you authored the thread.
> Reply to this email directly or view it on GitHub
>